components/worker/internal/bootstrap/config.go

```
@@ -62,20 +65,126 @@ type Config struct {
+			errs = append(errs, s.name+" must not use the default placeholder in production")
+		}
+	}
+
+	return errs
+}
+
 // InitWorker initializes and configures the application's dependencies and returns the Service instance.
-func InitWorker() *Service {
+// Uses a cleanup stack pattern: if any initialization step fails, all previously
+// opened connections are closed in reverse order to prevent resource leaks.
+func InitWorker() (_ *Service, err error) {
```

golangci-lint (2.5.0)

[error] cyclomatic complexity 19 of func `InitWorker` is high (> 16)

(gocyclo)